src/audio/base_fw.c

```diff
@@ -504,6 +504,28 @@ int schedulers_info_get(uint32_t *data_off_size,
 	return 0;
 }
 
+void set_perf_meas_state(const char *data)
+{
+	enum ipc4_perf_measurements_state_set state = *data;
+
+	struct telemetry_wnd_data *wnd_data =
+			(struct telemetry_wnd_data *)ADSP_DW->slots[DW_TELEMETRY_SLOT];
+	struct system_tick_info *systick_info =
+			(struct system_tick_info *)wnd_data->system_tick_info;
+
+	switch (state) {
```

Comment:
We probably want to print the state change for telemetry in the logs as it may have impact on performance or on issue reproducibility. i.e. a developer will know telemetry state when debugging.

Reply:
This made me think a bit. This IPC doesn't do much as of yet. It does some additional things in reference firmware, but nothing directly related to systick measurement (tests still use this one so we need it). It seems that systick measurement  in reference FW was on all the time, same here. This does not seem like it would take a lot of time but still...
I could modify this IPC and add a bool so that systick measurement is stopped unless it's triggered by this IPC.

Reply:
Thanks, this would just be informational to the developer. They would at least know if telemetry was on/off when an issue occurred.

Reply:
Got some more info, systick info measurement works all the time by design.  So there's no on/off switch. Still, what I can do is adding the log at the init of telemetry indicating that systick info measurement is active.